src/library_syscall.js

```diff
@@ -1688,9 +1663,12 @@ for (var x in SyscallsLibrary) {
     });
   }
   var pre = '', post = '';
-  if (which) {
+  if (which && t.indexOf(', varargs') != -1) {
     pre += 'SYSCALLS.varargs = varargs;\n';
   }
+  if (which && t.indexOf(', varargs') != -1) {
+    post += 'SYSCALLS.varargs = undefined;\n';
```

Comment:
how about only doing this in ASSERTIONS builds? It's only for assertion purposes I believe, and increases code size?

as a followup I assume we can get rid of `SYSCALLS.varargs` entirely, or will we always need it?

Reply:
Sure.